lib/ansible/modules/windows/win_cert_stat.ps1

```diff
@@ -0,0 +1,103 @@
+#!powershell
+
+# Copyright: (c) 2019, Micah Hunsberger
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+function ConvertTo-Timestamp($start_date, $end_date) {
+    if ($start_date -and $end_date) {
+        return (New-TimeSpan -Start $start_date -End $end_date).TotalSeconds
+    }
+}
+
+function Format-Date([DateTime]$date)
+{
+    return $date.ToUniversalTime().ToString('yyyy-MM-ddTHH:mm:ssK')
+}
+
+$store_location_values = ([System.Security.Cryptography.X509Certificates.StoreLocation]).GetEnumValues() | ForEach-Object { $_.ToString() }
+
+$spec = @{
+    options = @{
+        thumbprint = @{ type = "str"; required = $true }
+        store_name = @{ type = "str"; default = "My"; }
+        store_location = @{ type = "str"; default = "LocalMachine"; choices = $store_location_values; }
+    }
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$thumbprint = $module.Params.thumbprint
+$store_name = $module.Params.store_name
+$store_location = $module.params.store_location
+
+$module.Result.exists = $false
+
+$cert_path = [io.path]::Combine('Cert:\', $store_location, $store_name, $thumbprint)
+
+if (Test-Path -LiteralPath $cert_path)
```

Comment:
Using the PowerShell cert provider gives us a bit of a disconnect between the store names used in `win_certificate_store`. It would be nice if we can have both modules aligned.

Reply:
I agree, and I did originally use the x509store object, but the enum values were too limiting. For example, in a server 2012 instance, I can find 3 certificates in `cert:\localmachine\ca`, but I can't find any of them in the enum values of `AddressBook,AuthRoot,CertificateAuthority,Disallowed,My,Root,TrustedPeople,TrustedPublisher`

However, I _can_ open the 'CA' store using the `x509store` object, even though 'CA' isn't part of the enum.
```
$store = New-Object -TypeName X509Store -ArgumentList 'CA', "LocalMachine"
$store.Open("readOnly")
$store.Certificates # outputs the same as 'ls cert:\localmachine\ca'
```
I wasn't sure how to "test" that a store name existed, or how to generate a list of valid stores. Any ideas?

Reply:
Okay, turns out you can have arbitrary store names, and then create new stores that way as well. For that reason, if the store doesn't exist, trying to open it will result in 0 certificates, so I can switch to using the x509store object and still allow any string for `store_name`. If the store contains 0 certificates, would we prefer the module fail or just return an empty array?

Reply:
Oh wow, I did not know that. I think we should still document the valid enum stores that are available in the description but don't set it in the choices. Looks like `win_certificate_store` needs to be updated as well.

> If the store contains 0 certificates, would we prefer the module fail or just return an empty array?

I think just return an empty array is fine.